client/components/credit-card/stored-card.tsx

```diff
@@ -39,9 +44,19 @@ export const getCreditCardSummary = ( translate, type, digits ) => {
 	} );
 };
 
-const StoredCard = ( { lastDigits, cardType, name, expiry, translate, moment } ) => {
+interface Props {
+	lastDigits?: string;
+	cardType: string;
+	name: string;
+	expiry?: string;
+}
+
+const StoredCard: FunctionComponent< Props > = ( { lastDigits, cardType, name, expiry } ) => {
+	const translate = useTranslate();
+	const moment = useLocalizedMoment();
+
 	// The use of `MM/YY` should not be localized as it is an ISO standard across credit card forms: https://en.wikipedia.org/wiki/ISO/IEC_7813
-	const expirationDate = expiry ? moment( expiry ).format( 'MM/YY' ) : null;
+	const expirationDate = expiry ? moment( expiry, moment.ISO_8601, true ).format( 'MM/YY' ) : null;
```

Comment:
The Wikipedia link doesn't seem conclusive.

Do we have any need of localized moment if we're using static `MM/YY` format?

Reply:
Let's save that for a follow up 🙂 

From memory, the format I observed for `expiry` was `YYYY-MM`, but that may be inaccurate or accurate only under certain circumstances.